server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1609,6 +1634,11 @@ private static class IndexMetadataDiff implements Diff<IndexMetadata> {
                 indexWriteLoadForecast = null;
                 shardSizeInBytesForecast = null;
             }
+            if (in.getTransportVersion().onOrAfter(TransportVersions.EVENT_INGESTED_RANGE_IN_CLUSTER_STATE)) {
+                eventIngestedRange = IndexLongFieldRange.readFrom(in);
+            } else {
+                eventIngestedRange = IndexLongFieldRange.UNKNOWN;
+            }
```

Comment:
Can we add serialization test for index metadata that ensures we properly serialize the event ingested field - and also that bwc serialization works?

Reply:
I have a first pass at that in this new commit: https://github.com/elastic/elasticsearch/pull/106252/commits/5fb26b88e65792f79f7977ed4e2abf784060b0f6

I'm not clear on how exactly to structure a bwc test. I can say that earlier in the PR, several existing bwc (and mixed-cluster) tests were failing for several days while I tracked down all the places to set the eventIngested to UNKNOWN in order to get them passing again. I'd welcome any guidance on how to structure a unit test "bwc" test that focuses on this code.

Reply:
Revisiting this, I think fixing the todo you added in `IndexMetadataTests` should be enough. In order to do bwc testing, we'd have to manually serialize the indexmetadata object, which seems unnecessary given that we should be certain that any rolling upgrade tests will do so.

Reply:
Sounds good. New version of that test in next push.